django/core/serializers/xml_serializer.py

```diff
@@ -228,6 +234,11 @@ def _handle_object(self, node):
                     value = None
                 else:
                     value = field.to_python(getInnerText(field_node).strip())
+                    # Must treat special here since JSONField's to_python method
+                    # outputs string.
+                    if field_node.getAttribute("type") == "JSONField":
+                        value = json.loads(value)
+
```

Comment:
I agree that this solution is quite clunky :disappointed:, we could change `to_python()` and `value_to_string()` for `JSONField`, e.g.
```python
def to_python(self, value):
    if isinstance(value, str):
      value = json.loads(value)
    return value

def value_to_string(self, obj):
    return json.dumps(self.value_from_object(obj))
```
but it will break JSON serialization (see `TestSerialization.test_dumping` and  `TestSerialization.test_loading`).